pandas/core/resample.py

```diff
@@ -387,20 +387,28 @@ def _groupby_and_aggregate(self, how, grouper=None, *args, **kwargs):
 
         obj = self._selected_obj
 
+        result = None
         try:
             grouped = groupby(obj, by=None, grouper=grouper, axis=self.axis)
         except TypeError:
 
             # panel grouper
             grouped = PanelGroupBy(obj, grouper=grouper, axis=self.axis)
-
-        try:
             result = grouped.aggregate(how, *args, **kwargs)
```

Comment:
if you remove this (and remove the if statement for ``result is None`` on 399) does this stil work?

Reply:
@jreback 

This test is failed: `TestDatetimeIndex.test_resample_panel_numpy`
```
E       AssertionError: DataFrame.iloc[:, 0] are different
E       
E       DataFrame.iloc[:, 0] values are different (100.0 %)
E       [left]:  [nan, nan, nan, nan, nan, nan]
E       [right]: [-0.0517234265172, -0.475611549875, -0.32227637063, -0.443460761802, -0.0929628389341, -0.10035826564]
```

However applying these changes make tests to pass
```
--- a/pandas/core/resample.py
+++ b/pandas/core/resample.py
@@ -395,7 +395,11 @@ class Resampler(_GroupBy):
             grouped = PanelGroupBy(obj, grouper=grouper, axis=self.axis)
 
         try:
-            result = grouped.aggregate(how, *args, **kwargs)
+            if compat.callable(how) and not isinstance(grouped, PanelGroupBy):
+                # Check if the function is reducing or not.
+                result = grouped._aggregate_item_by_item(how, *args, **kwargs)
+            else:
+                result = grouped.aggregate(how, *args, **kwargs)
         except Exception:
```

Let me know, could I apply these chages or not?

Reply:
yeah let's go with your changes here. just trying to make code simpler.

Reply:
Excuse me, to be precise it should be:
```
--- a/pandas/core/resample.py
+++ b/pandas/core/resample.py
@@ -395,7 +395,11 @@ class Resampler(_GroupBy):
             grouped = PanelGroupBy(obj, grouper=grouper, axis=self.axis)
 
         try:
-            result = grouped.aggregate(how, *args, **kwargs)
+            if isinstance(obj, ABCDataFrame) and compat.callable(how):
+                # Check if the function is reducing or not.
+                result = grouped._aggregate_item_by_item(how, *args, **kwargs)
+            else:
+                result = grouped.aggregate(how, *args, **kwargs)
```

because `pandas.core.groupby.SeriesGroupBy` doesn't contain `_aggregate_item_by_item` method